portal-impl/src/com/liferay/portlet/documentlibrary/service.xml

```diff
@@ -514,6 +514,7 @@
 		<exception>InvalidFileEntryType</exception>
 		<exception>InvalidFileVersion</exception>
 		<exception>InvalidFolder</exception>
+		<exception>MismatchExtension</exception>
```

Comment:
Just a name change suggestion: Isn't `MismatchExtension` too generic? I think it would be better to name this `FileExtensionMismatchException`, to make it more apparent what the error is about.